src/Microsoft.TemplateEngine.Cli/CommandParsing/NewCommandInputCli.cs

```diff
@@ -287,7 +287,7 @@ public string TemplateParamInputFormat(string canonical)
         {
             foreach (string variant in VariantsForCanonical(canonical))
             {
-                if (_parseResult.Tokens.Contains(variant))
+                if (_parseResult.Tokens.Any(s => s.Contains(variant)))
```

Comment:
Don't we run into a problem here? 
What happens if we have two options and one is a substring of another? 
There also may be problems when the option is written like `--option value` because it parses as two separate tokens.

Reply:
Good catch. I've replaced it with search for s.StartsWith($"{variant}=") as the fallback. 

However, imo this should be fixed in Tokens collection in Microsoft.DotNet.Cli.CommandLine.dll.  When parsing the arguments we supply in configuration to use "=" delimiter, however it is not reflected in Tokens collection, and parameter used as `--aaa=bbb`, is returned as single token. In all other cases --aaa and bbb are returned as separate tokens.  It is possible to get exact information on Tokens used inside AppliedOptions collection, but it's quite deep inside.  It still possible to get use of this information but I would estimate the fix in 2d/work, which in my opinion just not justify the effort to dig in this information to show only in this case of error.  

Also I'm checking with Kathleen now if she is ok with the wording for the error message. It might be that we don't need the information at all.